src/knapsack/MultipleKnapsackModel.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
+# ALL RIGHTS RESERVED
+"""
+Created on February 7, 2021
+
+@author: mandd
+"""
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import math
+import copy
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from PluginsBaseClasses.ExternalModelPluginBase import ExternalModelPluginBase
+from utils import InputData, InputTypes
+#Internal Modules End-----------------------------------------------------------
+
+
+class MultipleKnapsackModel(ExternalModelPluginBase):
+  """
+    This class is designed to create the MultipleKnapsack model
+  """
+
+  @classmethod
+  def getInputSpecs(cls):
+    """
+      Collects input specifications for this class.
+      @ In,  None
+      @ Out, inputSpecs, InputData, input specifications
+    """
+    inputSpecs = InputData.parameterInputFactory('ExternalModel')
+    inputSpecs.addParam('name'   , param_type=InputTypes.StringType, required=True)
+    inputSpecs.addParam('subType', param_type=InputTypes.StringType, required=True)
+
+    inputSpecs.addSub(InputData.parameterInputFactory('penaltyFactor', contentType=InputTypes.FloatType))
+    inputSpecs.addSub(InputData.parameterInputFactory('outcome'      , contentType=InputTypes.StringType), required=True)
+    inputSpecs.addSub(InputData.parameterInputFactory('choiceValue'  , contentType=InputTypes.StringType), required=True)
+
+    knapsack = InputData.parameterInputFactory('knapsack', contentType=InputTypes.StringType)
+    knapsack.addParam('ID', param_type=InputTypes.StringType, required=True)
+    inputSpecs.addSub(knapsack)
+
+    mapping = InputData.parameterInputFactory('map', contentType=InputTypes.StringType)
+    mapping.addParam('value', param_type=InputTypes.StringType, required=True)
+    mapping.addParam('cost',  param_type=InputTypes.StringType, required=True)
+    inputSpecs.addSub(mapping)
```

Comment:
If you inherit directly from BaseKnapsackModel, most of these lines can be removed